src/libs/actions/Report.ts

```diff
@@ -3714,6 +3715,30 @@ function setGroupDraft(newGroupDraft: Partial<NewGroupChatDraft>) {
     Onyx.merge(ONYXKEYS.NEW_GROUP_CHAT_DRAFT, newGroupDraft);
 }
 
+function buildOptimisticTaskDataForNewAssingee(assigneeLogin: string) {
+    const assigneeAccountID = UserUtils.generateAccountID(assigneeLogin);
+    const report: OnyxEntry<Report> | undefined = ReportUtils.buildOptimisticChatReport([assigneeAccountID]);
+    report.isOptimisticReport = true;
+
+    // When assigning a task to a new user, by default we share the task in their DM
+    // However, the DM doesn't exist yet - and will be created optimistically once the task is created
+    // We don't want to show the new DM yet, because if you select an assignee and then change the assignee, the previous DM will still be shown
+    // So here, we create it optimistically to share it with the assignee, but we have to hide it until the task is created
+    if (report) {
+        report.isHidden = true;
+    }
+    Onyx.set(`${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`, report);
+
+    const optimisticPersonalDetailsListAction = {
+        accountID: assigneeAccountID,
+        avatar: FallbackAvatar,
+        displayName: assigneeLogin,
+        login: assigneeLogin,
+    };
+    Onyx.merge(ONYXKEYS.PERSONAL_DETAILS_LIST, {[assigneeAccountID]: optimisticPersonalDetailsListAction});
+    return {assignee: optimisticPersonalDetailsListAction, assigneeReport: report};
```

Comment:
This logic is same on `setAssigneeValue`. Let's use this function there

Reply:
this function has other logic like set `assigneeChatReport`, `setShareDestinationValue` so I don't think we should use this function here.

Reply:
Yes. What I meant is to use that function in `setAssigneeValue` (and not to replace it) so we don't write duplicate code.

Reply:
@s77rt I updated.